### PR TITLE
Lesson 5-7 p3: test script

### PR DESCRIPTION
**Pull request by a-vodka**
Test script for character device
please look at screenshot

image

### CODE REVIEW

**Comment by galkinletter** [comment]
Makefile - Need  fix var for  /home/alex/Downloads/linux-5.0.7 and etc
Makefile -   Need fix  end of line  

chrdev.c - Fix indents. Different indents

**Review by galkinletter — CHANGES_REQUESTED** [review]
Need fix Makefile and chrdev.c 

See comment

**Comment by galkinletter** [comment]
Makefile - Please, add empty line at the end of file

File: 02-KernelModules_CharacterDevice/Makefile [hunk 1]
```diff
@@ -0,0 +1,7 @@
+obj-m += chrdev.o
+
+all:
+	make -C /home/alex/Downloads/linux-5.0.7 M=$(PWD) modules
+
+clean:
+	make -C /lib/modules/$(shell uname -r)/build M=$(PWD) clean
```
Comment (galkinletter) [inline 02-KernelModules_CharacterDevice/Makefile]:
```suggestion
	make -C /lib/modules/$(shell uname -r)/build M=$(PWD) clean
	
```

Makefile - Need fix end of line

File: 02-KernelModules_CharacterDevice/chrdev.c [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+}
+
+module_init(chrdev_init);
+module_exit(chrdev_exit);
+
+
+MODULE_AUTHOR("Oleksii.Vodka <oleksii.vodka@gmail.com>");
+MODULE_DESCRIPTION("Character device driver");
+MODULE_LICENSE("GPL");
+MODULE_VERSION("0.1");
```
Comment (galkinletter) [inline 02-KernelModules_CharacterDevice/chrdev.c]: Need fix end of line

File: 02-KernelModules_CharacterDevice/test-chr-dev.sh [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+echo "Device status..."
+cat /proc/vdk-chrdev-example
+
+rmmod $MOD
+
+insmod $MODULE_PATH buffer_size=128
+echo "Device status..."
+cat /proc/vdk-chrdev-example
+rmmod $MOD
```
Comment (galkinletter) [inline 02-KernelModules_CharacterDevice/test-chr-dev.sh:35]:
```suggestion
rmmod $MOD

```

Need fix end of line

**Review by galkinletter — CHANGES_REQUESTED** [review]
Need fix

File: 02-KernelModules_CharacterDevice/chrdev.c [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+		return PTR_ERR(pdev);
+	}
+	pr_info("chrdev: device node created successfully\n");
+
+	example_kobject = kobject_create_and_add(DEVICE_NAME, kernel_kobj);
+    if(!example_kobject) {
+        pr_debug("failed to create kobject \n");
+        return -ENOMEM;
+    }
+    error = sysfs_create_file(example_kobject, &sysfs_attribute.attr);
```
Comment (oleg-h2o) [inline 02-KernelModules_CharacterDevice/chrdev.c:222]: Error handling is broken here.

**Review by oleg-h2o — CHANGES_REQUESTED** [review]
Fix error handling.

**Review by galkinletter — CHANGES_REQUESTED** [review]
Please Fix error handling.

File: 02-KernelModules_CharacterDevice/chrdev.c (hunk 4 above)
Comment (a-vodka) [reply 02-KernelModules_CharacterDevice/chrdev.c:222]: @oleg-h2o 

> Error handling is broken here.


What is it mean?

**Pull request merged**